Admin/Test-AMSI.ps1

```diff
@@ -1,144 +1,793 @@
+                        } else {
+                            Write-Warning "We did not find HttpRequestFilteringModule on ClientAccess ECP web.config"
+                            Write-Warning "Path: $($ExchangePath)ClientAccess\ecp\web.config"
+                        }
+                    } else {
+                        Write-Warning "We did not find web.config for ClientAccess ECP"
+                        Write-Warning "Path: $($ExchangePath)ClientAccess\ecp\web.config"
+                    }
+                } else {
+                    Write-Host "We could not get FrontEnd or BackEnd Web.config path on $ExchangeServer." -ForegroundColor Red
+                }
+            } else {
+                Write-Host "Cannot get Exchange installation path on $server" -ForegroundColor Red
+            }
+
+            Write-Host $msgNewLine
+            Write-Host "AMSI is Enabled on Server $ExchangeServer." -ForegroundColor Green
+            Write-Host "We did not find any Settings Override that remove AMSI on server $ExchangeServer."
+            Write-Host ""
+        }
     }
-    if ($DisableAMSI) {
-        New-SettingOverride -Name DisablingAMSIScan -Component Cafe -Section HttpRequestFiltering -Parameters ("Enabled=False") -Reason "Disabled via CSS-Exchange Script"
-        Get-ExchangeDiagnosticInfo -Process Microsoft.Exchange.Directory.TopologyService -Component VariantConfiguration -Argument Refresh
-        Write-Warning "Remember to restart IIS for this to take affect. You can accomplish this by running .\Test-AMSI.ps1 -RestartIIS"
-        return
+
+    $msgNewLine = "`n"
+    $bar = ""
+    1..($Host.UI.RawUI.WindowSize.Width) | ForEach-Object { $bar += "-" }
+    Write-Host ""
+    Write-Host $bar
+    if (-not (Confirm-Administrator)) {
+        Write-Host $msgNewLine
+        Write-Warning "This script needs to be executed in elevated mode. Start the Exchange Management Shell as an Administrator and try again."
+        exit
     }
-    if ($RestartIIS) {
-        Restart-Service -Name W3SVC, WAS -Force
-        return
+
+    $exchangeShell = Confirm-ExchangeShell
+    if (-not($exchangeShell.ShellLoaded)) {
+        Write-Host $msgNewLine
+        Write-Warning "Failed to load Exchange Shell Module..."
+        exit
     }
 }
 
-Test-AMSI
+process {
+    $filterList = @()
+
+    if ( $PSCmdlet.ParameterSetName -eq "TestAMSI" -or $PSCmdlet.ParameterSetName -eq "TestAMSIAll" ) {
+        $TestAMSI = $true
+    }
+
+    if ( $null -eq $ServerList -and ( $RestartIIS -or $CheckAMSIConfig -or $TestAMSI ) -and ( -not $AllServers ) ) {
+        $ServerList = $env:COMPUTERNAME
+    }
+
+    Write-Host ""
+    Write-Host "AMSI only support on Exchange 2016 CU21 or newer, Exchange 2019 CU10 or newer and running on Windows 2016 or newer" -ForegroundColor Green
+    Write-Host ""
+
+    $SupportedExchangeServers = Get-ExchangeServer | Where-Object {
```

Comment:
Just use the shared function `Get-ExchangeBuildVersionInformation.ps1` to query the build information and use them to find out if we are running the correct version:

![image](https://github.com/microsoft/CSS-Exchange/assets/40993616/1e96857c-8517-484e-8958-259e1f85ae7c)
![image](https://github.com/microsoft/CSS-Exchange/assets/40993616/baf13d09-d6df-43c3-945e-fe017d13cb7d)
![image](https://github.com/microsoft/CSS-Exchange/assets/40993616/ad0c6412-fb31-43c8-b92b-bb0dc11109eb)